widget/playground/src/hooks/useFontLoader.ts

```diff
@@ -0,0 +1,40 @@
+import { useState } from 'react';
+
+import { FONTS } from '../constants';
+
+const useFontLoader = () => {
+  const [fontLink, setFontLink] = useState<HTMLLinkElement | null>(null);
+
+  const loadFont = (fontUrl: string) => {
+    const link = document.createElement('link');
+    link.href = fontUrl;
+    link.rel = 'stylesheet';
+    document.head.appendChild(link);
+    return link;
+  };
+
+  const unloadFont = () => {
+    if (fontLink) {
+      document.head.removeChild(fontLink);
+      setFontLink(null);
+    }
+  };
+
+  const handleLoadCustomFont = (fontName: string) => {
+    unloadFont(); // Remove previous font
+    const fontUrl = getFontUrl(fontName);
+    if (fontUrl) {
+      const newFontLink = loadFont(fontUrl);
+      setFontLink(newFontLink);
+    }
+  };
+
+  const getFontUrl = (fontName: string) => {
```

Comment:
I think move this out of hook is better, you can move it to `utils` as well since it's a pure function.

Reply:
Please also rename the function to `getFontUrlByName`

Reply:
Don't forget to type the return.